src/Codec/Beam.hs

```diff
@@ -230,6 +240,15 @@ appendTerm builder term =
               { atomTable = Map.insert name value old
               }
 
+    withLiteral literal toBuilder =
+      let
+        new = litTable builder ++ [literal]
```

Comment:
What if we prepend to the table then reverse it in `encodeLiterals`? That would definitely be cheaper